packages/blade/src/hooks/useWindowSize.ts

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+interface Size {
+  width: number;
+  height: number;
+}
+
+function useWindowSize(): Size {
+  // Initialize state with undefined width/height so server and client renders match
+  // Learn more here: https://joshwcomeau.com/react/the-perils-of-rehydration/
+  const [windowSize, setWindowSize] = React.useState<Size>({
+    width: window.innerWidth,
+    height: window.innerHeight,
+  });
+  React.useEffect(() => {
+    // Handler to call on window resize
+    function handleResize(): void {
+      // Set window width/height to state
+      setWindowSize({
+        width: window.innerWidth,
+        height: window.innerHeight,
+      });
+    }
+    // Add event listener
+    window.addEventListener('resize', handleResize);
```

Comment:
Probably fine, height to zayada change nehi hoga, plus since bottomsheet will only render on mobile devices there won't be any further resizes one this value is populated.